src/solver/optimisation/opt_period_string_generator_base.h

```diff
@@ -37,4 +38,8 @@ class OptPeriodStringGenerator
 public:
     virtual std::string to_string() const = 0;
     virtual ~OptPeriodStringGenerator() = default;
+    std::string createOptimizationFilename(const std::string& title,
```

Comment:
Base class **OptPeriodStringGenerator** is an abstraction for the optimization period, that is either :  
- \<year\>-\<week\>
- \<year\>-\<week\>-\<day\>

inside the filename.
So why do we have a new **createOptimizationFilename** function that take many arguments about the whole file name ?
I guess **OptPeriodStringGenerator** should concern the optimization period only.